tests/Feature/Platform/Services/UserAgentServiceTest.php

```diff
@@ -318,4 +356,64 @@ public function testDolphinAlphaUserAgent(): void
             'integrationVersion' => '1.2.1.47-alpha',
         ], $this->parseUserAgent($userAgent));
     }
+
+    public function testDolphinUserAgent(): void
+    {
+        $userAgent = 'Dolphin/2407-123';
+
+        $this->assertEquals([
+            'client' => 'Dolphin',
+            'clientVersion' => '2407-123',
+        ], $this->parseUserAgent($userAgent));
+
+        $userAgent = 'Dolphin/2407-171-dirty';
+
+        $this->assertEquals([
+            'client' => 'Dolphin',
+            'clientVersion' => '2407-171-dirty',
+        ], $this->parseUserAgent($userAgent));
+
+        $userAgent = 'Dolphin/5.0-21878';
+
+        $this->assertEquals([
+            'client' => 'Dolphin',
+            'clientVersion' => '5.0-21878',
+        ], $this->parseUserAgent($userAgent));
+
+        // dolphin changed user agent from 5.0-build to YYMM-build in July 2024
+        $this->assertEquals(-1, UserAgentService::versionCompare('2407-123', '2407-123-dirty')); // 2407-123 < 2407-123-dirty
+        $this->assertEquals(-1, UserAgentService::versionCompare('2407-5', '2409-17')); // 2407-5 < 2409-17
+        $this->assertEquals(1, UserAgentService::versionCompare('2409-67', '5.0-21878')); // 2409-67 > 5.0-21878
+    }
+
+    public function testFlycastUserAgent(): void
+    {
+        $userAgent = 'Flycast/45bf218df CFNetwork/1498.700.2 Darwin/23.6.0';
+
+        $this->assertEquals([
+            'client' => 'Flycast',
+            'clientVersion' => '45bf218df',
+            'extra' => [
+                'CFNetwork' => '1498.700.2',
+                'Darwin' => '23.6.0',
+            ],
+        ], $this->parseUserAgent($userAgent));
+
+         // we can't enforce minimum versions on git hashes
+         $this->assertEquals(0, UserAgentService::versionCompare('45bf218df', '2d4684a46'));
```

Comment:
Is the intention to treat all Git hashes as equal versions?

For example:
```php
$hashOne = "abcdef1";
$hashTwo = "1234567";

$result = UserAgentService::versionCompare($hashOne, $hashTwo);

$this->assertEquals(0, $result); // FAILS
```
I could be misinterpreting the code though.

Reply:
hashTwo is completely numeric, so it's considered version "1234567". hashOne is alphanumeric, so it's ignored. As a result, the "unknown" version is considered older than the "1234567" version.

Reply:
Ah, that makes sense.